### PR TITLE
Hide dm patient days

### PR DESCRIPTION
**Pull request by harimohanraj89**
**Story card:** 
* https://app.shortcut.com/simpledotorg/story/9179
* https://app.shortcut.com/simpledotorg/story/9178/hide-patient-days-from-csv-download
* https://app.shortcut.com/simpledotorg/story/9177/hide-patient-days-column-for-diabetes-drug-stock-section

## Because

West Bengal has added DM drugs to drug stock, but there aren't good coefficients to calculate patient days yet.

## This addresses

Hides patient days for diabetes drugs for now. Once countries start coming up with coefficients, we can switch this logic to instead hide patient days when coeffs are not configured. For now, this unblocks WB faster, and we don't know if anyone else will use DM drug stocks yet.

We also move DM drugs to be the last section in the drug stock report. This PR achieves that by:
* Adding "diabetes" to the custom country config drug category order
* Updating the ordering logic to be applied if the reporting categories are a _subset_ of the custom order, not necessarily exactly equal to the custom order.

Screen Shot 2022-10-19 at 11 13 13 AM

## Test instructions

Create Diabetes drugs in a medication protocol
* Set category to "Diabetes"
* Set Stock Tracked to "Yes"
View a drug stock report for an area using the protocol
Create some drug stocks, entering DM drug info as well
Ensure that
* Patient days columns are not shown for DM drugs in dashboard
* Patient days columns are not shown for DM drugs in download
* DM drugs are the last section
* All column headers and datapoints line up correctly in tables and downloads

### CODE REVIEW

File: app/queries/drug_stocks_query.rb [hunk 1]
```diff
@@ -76,7 +76,7 @@ def drug_consumption_report
 
     custom_drug_category_order = CountryConfig.current.fetch(:custom_drug_category_order, [])
 
-    if custom_drug_category_order.to_set == drugs_by_category.keys.to_set
+    if (drugs_by_category.keys - custom_drug_category_order).empty?
```
Comment (harimohanraj89) [inline app/queries/drug_stocks_query.rb:79]: Apply the ordering even if the categories are a subset of the custom order list. This way, West Bengal can track DM drugs and leverage this order AND states that are not tracking DM drugs will still be able to leverage the hypertension drugs order.

File: app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb [hunk 2]
```diff
@@ -3,9 +3,12 @@
...
       <col>
-      <% @drugs_by_category.map do |_drug_category, drugs| %>
-        <col class="table-divider">
-        <% drugs.map do |_drug| %>
+      <% @drugs_by_category.map do |drug_category, drugs| %>
+        <% drugs.each_with_index.map do |_drug, index| %>
+          <col class="<%= "table-divider" if index.zero? %>">
+        <% end %>
+
+        <% unless drug_category == "diabetes" %>
```
Comment (prabhanshuguptagit) [inline app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb:11]: Let's extract this to a helper method like `show_patient_days?`.

File: app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb [hunk 3]
```diff
@@ -51,17 +60,19 @@
...
-              data-original-title="<%= render "drug_stocks_tooltip", report: @report.dig(:total_patient_days, drug_category) %>"
-              data-sort-column-key=<%= "#{drug_category}_patient_days" %>>
-            <em class="<%= patient_days_css_class(patient_days) %>">
-              <%= patient_days %>
-            </em>
-          </td>
-        <% else %>
-          <td class="type-blank"><span>&#8212;</span></td>
+        <% unless drug_category == "diabetes" %>
+          <% if patient_days.present? %>
```
Comment (prabhanshuguptagit) [inline app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb:64]: [nit] `if show_patient_days?(drug_category) && patient_days.present?`

File: app/exporters/drug_stocks_report_exporter.rb [hunk 4]
```diff
@@ -33,17 +33,20 @@ def drug_categories_header
     left_padding_columns = [nil] * left_pad_size
 
     left_padding_columns + @drugs_by_category.flat_map do |category, drugs|
-      [protocol_drug_labels[category][:full], [nil] * drugs.count].flatten
+      category_column_count = show_patient_days?(category) ? drugs.count : drugs.count - 1
```
Comment (prabhanshuguptagit) [inline app/exporters/drug_stocks_report_exporter.rb:36]: Something might be up with this calculation, `Column M` should say diabetes
image

File: app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb [hunk 5]
```diff
@@ -3,9 +3,12 @@
     <colgroup>
```
Comment (prabhanshuguptagit) [inline app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb:3]: Praveen mentioned that the layout breaks if there's too many categories, can we check if it's addressed? https://simpledotorg.slack.com/archives/C03URFX2LUT/p1666007119061689?thread_ts=1665984305.235149&cid=C03URFX2LUT

**Review by prabhanshuguptagit — APPROVED** [review]
Looks great, approving to unblock.

**Comment by prabhanshuguptagit** [comment]
@harimohanraj89 I've tacked on a change I had to make to seed drug stocks in the last month, since we don't show drug stock reports for the current month and the reports are all empty. [3beff00](https://github.com/simpledotorg/simple-server/pull/4323/commits/3beff00c7a6e6f873b65da77ab933f788d74e96d)

File: app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb (hunk 2 above)
Comment (harimohanraj89) [reply app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb:11]: Good call, similar to the CSV exporter.

File: app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb (hunk 3 above)
Comment (harimohanraj89) [reply app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb:64]: That's what I thought first, but the `patient_days.present?` conditional has an `else` below.

File: app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb (hunk 5 above)
Comment (dburka) [reply app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb:3]: It's sort-of addressed @prabhanshuguptagit ... on narrow screens or if you add too many drug/dosage combinations to track, it still runs over the side of the card. This is not a blocker.

**Review by dburka — APPROVED** [review]
LGTM

File: app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb (hunk 2 above)
Comment (harimohanraj89) [reply app/views/my_facilities/drug_stocks/_drug_stocks_table.html.erb:11]: Donezo.

File: app/exporters/drug_stocks_report_exporter.rb (hunk 4 above)
Comment (harimohanraj89) [reply app/exporters/drug_stocks_report_exporter.rb:36]: @prabhanshuguptagit where'd you see this? Local test data? Is Metformin set as a diuretic drug?

Comment (prabhanshuguptagit) [reply app/exporters/drug_stocks_report_exporter.rb:36]: Oh whoops. I think I was going back and forth changing Metmorfin's type on local test data yes. Not a bug.

image

**Pull request merged**